C/Coin-OR/CSDP/bundled/patches/blegat.patch

```diff
@@ -0,0 +1,1731 @@
++    printf("block[%d]: %p\n", blk, &A.blocks[blk]);
++    printb(A.blocks[blk]);
++  }
++
++}
++
++FILE *fid;
++
++void print_sparse_block(struct sparseblock *b) {
++  int i;
++  fid = stdout;
++  fprintf(fid, "\n* Printing block: %p\n", b);
++  if (b == NULL)
++    return;
++  fprintf(fid, " next: %p\n", b->next);
++  fprintf(fid, " nextbyblock: %p\n", b->nextbyblock);
++  fprintf(fid, " constraintnum: %d\n", b->constraintnum);
++  fprintf(fid, " blocknum: %d\n", b->blocknum);
++  fprintf(fid, " blocksize: %d\n", b->blocksize);
++  fprintf(fid, " numentries: %d\n", b->numentries);
++  if (b->blocksize <= 30) {
++    for (i = 1; i <= b->numentries; i++) {
++      fprintf(fid, "  block[%d, %d] = %f\n",
++              b->iindices[i],
++              b->jindices[i],
++              b->entries[i]);
++    }
++  } else {
++    fprintf(fid, " TOO LARGE\n");
++  }
++  print_sparse_block(b->next);
++}
++
++void  print_constraints(int k,
++                        struct constraintmatrix *constraints)
++{
++  int i, j;
++  fid = stdout;
++  struct sparseblock *p;
++
++  fprintf(fid, "constraints == %p\n", constraints);
++
++  for (i=1; i<=k; i++)
++    {
++      fprintf(fid, "\n\nprinting constraints[%d].blocks\n", i);
++      p=constraints[i].blocks;
++      while (p != NULL)
++        {
++          fprintf(fid, "p == %p\n\n", p);
++          print_sparse_block(p);
++          fprintf(fid, "\n");
++          for (j=1; j<=p->numentries; j++)
++            {
++              fprintf(fid, "i=%d, j=%d\n", i, j);
++              fprintf(fid,"%d %d %d %d %f \n",
++                      i,
++                      p->blocknum,
++                      p->iindices[j],
++                      p->jindices[j],
++                      p->entries[j]);
++            };
++          p=p->next;
++        };
++    };
++}
+--- a/lib/easysdp.c
++++ b/lib/easysdp.c
+@@ -25,9 +25,32 @@
+ int easy_sdp(n,k,C,a,constraints,constan
+      double *ppobj;
+      double *pdobj;
+ {
++   struct paramstruc params;
++   int printlevel;
++   /*
++    *  Initialize the parameters.
++    */
++   initparams(&params,&printlevel);
++   return parametrized_sdp(n,k,C,a,constraints,constant_offset,pX,py,pZ,ppobj,pdobj,printlevel,params);
++}
++
++int parametrized_sdp(n,k,C,a,constraints,constant_offset,pX,py,pZ,ppobj,pdobj,printlevel,params)
++     int n;
++     int k;
++     struct blockmatrix C;
++     double *a;
++     struct constraintmatrix *constraints;
++     double constant_offset;
++     struct blockmatrix *pX;
++     double **py;
++     struct blockmatrix *pZ;
++     double *ppobj;
++     double *pdobj;
++     int printlevel;
++     struct paramstruc params;
++{
+   int ret;
+   struct constraintmatrix fill;
+-  struct paramstruc params;
+   struct blockmatrix work1;
+   struct blockmatrix work2;
+   struct blockmatrix work3;
+@@ -53,7 +76,6 @@ int easy_sdp(n,k,C,a,constraints,constan
+   double *dy1;
+   double *rhs;
+   double *besty;
+-  int printlevel;
```

Comment:
It seems there is an issue with this patch. This removal is not applied and it is causing the compilation failure. I tried locally to apply this patch and it is also not removed.

Reply:
I made a PR in https://github.com/amontoison/Yggdrasil/pull/1 to fix this

Reply:
Thanks blegat! I must have made a mistake when I generated the patch. All checks have passed :rocket: